main.go

```diff
@@ -265,6 +271,23 @@ func main() {
 	<-ctx.Done()
 }
 
+func getSriovTokenServerChainElement(ctx context.Context) (tokenServer networkservice.NetworkServiceServer) {
+	sriovTokens := tokens.FromEnv(os.Environ())
+	switch len(sriovTokens) {
+	case 0:
+		tokenServer = null.NewServer()
+	case 1:
+		var tokenKey string
+		for tokenKey = range sriovTokens {
+			break
+		}
+		tokenServer = token.NewServer(tokenKey)
+	default:
+		log.FromContext(ctx).Fatalf("endpoint must be configured with none or only one sriov resource")
+	}
+	return
+}
```

Comment:
@edwarnicke, @Bolodya1997 , @pperiyasamy 

Can we use here switch-case chain element?

Take a look at examples in this PR https://github.com/networkservicemesh/sdk/pull/1046

Reply:
@denis-tingaikin thanks! switch-case chain element seems interesting, but the predicate [arguments](https://github.com/networkservicemesh/sdk/blob/main/pkg/networkservice/common/switchcase/default.go#L26) is made to use ctx and conn objects which is not the case here, so is it still ok to use switch-case chain element here ?

Reply:
I disagree:
1. `switchcase` is designed to route Request to the correct chain by some request-time condition. If we have condition resolved before the time of building chain, we can just put right server into the chain.
2. I don't actually think that using swithcase will make code here more short/readable/etc, because we need to have an instance of the server **before** the condition. So it actually will look like:
```go
sriovTokens := tokens.FromEnv(os.Environ())

var tokenServer networkservice.Server
if len(sriovTokens) == 1 {
	var tokenKey string
	for tokenKey = range sriovTokens {
		break
	}
	tokenServer = token.NewServer(tokenKey)
}

endpoint := ...
	switchcase.NewServer([]...{
		{
			Condition: func(...) bool { return len(sriovTokens) == 0 }
			Server: null.NewServer(),
		},
		{
			Condition: func(...) bool { len == 1 },
			Server: tokenServer,
		},
		{
			Condition: switchcase.Default(),
			Server: injecterror.NewServer(...),
		},
	})
	...
```

Reply:
> 1. `switchcase` is designed to route Request to the correct chain by some request-time condition.

+1

Reply:
got it, thanks :)